Godeps/_workspace/src/k8s.io/kubernetes/pkg/storage/etcd/etcd_helper.go

```diff
@@ -264,6 +264,20 @@ func (h *etcdHelper) Delete(ctx context.Context, key string, out runtime.Object)
 }
 
 // Implements storage.Interface.
+func (h *etcdHelper) DeleteDir(ctx context.Context, directory string) error {
+	if ctx == nil {
+		glog.Errorf("Context is nil")
```

Comment:
you're right... the other places are weird, too

if ctx is required, we should return an error
if it's not required, we shouldn't log an error